src/mongocxx/collection.cpp

```diff
@@ -426,12 +426,89 @@ std::int64_t collection::count(bsoncxx::document::view filter, const options::co
 }
 
 bsoncxx::document::value collection::create_index(bsoncxx::document::view keys,
-                                                  bsoncxx::document::view options) {
+                                                  const options::index& options) {
     scoped_bson_t bson_keys{keys};
+    scoped_bson_t weights{options.weights()};
+    scoped_bson_t partial_filter_expression{options.partial_filter_expression()};
     bson_error_t error;
+    mongoc_index_opt_geo_t geo_opt{};
```

Comment:
just to be paranoid, can you prefix all mongoc symbols with `::` (global namespace resolution operator)

e.g.

`::mongoc_index_opt_geo_t geo_opt{}`